packages/zipkin/test/internalRepresentations.test.js

```diff
@@ -176,3 +179,49 @@ describe('JSON Formatting', () => {
     ]);
   });
 });
+
+describe('64-bit trace ID', () => {
+  const serverSpan = new MutableSpan(new TraceId({
+    traceId: new Some('000000000000162e'),
+    spanId: '000000000000162e',
+    sampled: None
+  }));
+
+  it('should opaquely write trace ID in JSON representation', () => {
+    const spanJson = serverSpan.toJSON();
+    expect(spanJson.traceId).to.equal(serverSpan.traceId.traceId);
+  });
+
+  it('should write trace ID as high and low numbers in Thrift representation', () => {
+    const serialized = serializeSpan(serverSpan, format = 'raw');
+
+    const spanThrift = new thriftTypes.Span();
+    spanThrift.read(new TBinaryProtocol(new TFramedTransport(serialized)));
+
+    expect(spanThrift.trace_id_high).to.equal(null);
+    expect(spanThrift.trace_id.toNumber(true)).to.equal(5678);
+  });
+});
+
+describe('128-bit trace ID', () => {
+  const serverSpan = new MutableSpan(new TraceId({
+    traceId: new Some('00000000000004d2000000000000162e'),
+    spanId: '000000000000162e',
+    sampled: None
+  }));
+
+  it('should opaquely write trace ID in JSON representation', () => {
+    const spanJson = serverSpan.toJSON();
+    expect(spanJson.traceId).to.equal(serverSpan.traceId.traceId);
+  });
+
+  it('should write trace ID as high and low numbers in Thrift representation', () => {
+    const serialized = serializeSpan(serverSpan, format = 'raw');
+
+    const spanThrift = new thriftTypes.Span();
+    spanThrift.read(new TBinaryProtocol(new TFramedTransport(serialized)));
+
+    expect(spanThrift.trace_id_high.toNumber(true)).to.equal(1234);
```

Comment:
Yeah, this is annoying with thrift :(. node-int64 does not work as a number even if they claim to do so.